cwl/workflow/data_assembly_DNA_wf.cwl

```diff
@@ -0,0 +1,92 @@
+cwlVersion: v1.0
+class: Workflow
+id: data_assembly_DNA_wf
+requirements:
+  - class: MultipleInputFeatureRequirement
+  - class: SubworkflowFeatureRequirement
+  - class: ScatterFeatureRequirement
+inputs:
+  # maf
+  input_maf: { type: File }
+  input_previous_merged_maf: { type: File }
+
+  
+  # CNV
+  input_SV: { type: File? }
+  input_cnvkit_call_cns: { type: File }
+  input_cnvkit_call_seg: { type: File }
+  input_controlfreeC_p_value: { type: File}
+  input_controlfreeC_info: { type: File }
+
+
+  participant_id: string
+  biospecimen_id_tumor: string
+  biospecimen_id_normal: string
+
+  run_WGS: { type: boolean, doc: "require run WGS with SV" }
+
+
+outputs:
+  new_merged_maf: { type: File, outputSource: merge_maf/output_merged_maf}
+
+  formatted_cnvkit: { type: File, outputSource: format_cnvkit_cnv/output_formatted_cnvkit }
+  formatted_controlfreec: { type: File, outputSource: format_controlfreeC_cnv/output_formatted_controlfreeC }
+  formatted_sv: { type: File, outputSource: format_annoSV/output_formatted_annoSV }
+
+steps:
+  gatekeeper:
+    run: ../tools/gatekeeper.cwl
+    in:
+      run_WGS: run_WGS
+    out: [scatter_WGS]
+
+  merge_maf:
+    run: ../tools/merge_maf.cwl
+    in:
+      input_maf: input_maf
+      input_previous_merged_maf: input_previous_merged_maf
+      biospecimen_id: biospecimen_id_tumor
+
+    out: [output_merged_maf]
+  
+  annoSV:
+    run: ../tools/annoSV.cwl
+    in:
+      input_SV: input_SV
+      conditional_run: gatekeeper/scatter_WGS
+    scatter: conditional_run
+    out: [output_formatted_SV]
```

Comment:
When running without args, cwltool throws error:
```
cwl/workflow/data_assembly_DNA_wf.cwl:58:11: Source 'output_formatted_SV' of type {"type": "array",
                                             "items": "File"} is incompatible
cwl/workflow/data_assembly_DNA_wf.cwl:63:7:     with sink 'input_annoSV' of type "File"
cwl/workflow/data_assembly_DNA_wf.cwl:69:11: Source 'output_formatted_annoSV' of type {"type":
                                             "array", "items": "File"} is incompatible
cwl/workflow/data_assembly_DNA_wf.cwl:34:19:   with sink 'formatted_sv' of type "File"
```

Reply:
also fixed :
should be good now, I will run a test run with the SV data 
```
➜  workflow git:(cwl_workflow) ✗ cwltool data_assembly_DNA_wf.cwl
INFO /Users/zhangb1/miniconda3/bin/cwltool 3.0.20200807132242
INFO Resolved 'data_assembly_DNA_wf.cwl' to 'file:///Users/zhangb1/Documents/Kid-First/data_assembly/branch/d3b-bixu-data-assembly/cwl/workflow/data_assembly_DNA_wf.cwl'
data_assembly_DNA_wf.cwl:90:5: Warning: checking item
                               Warning:   Field `class` contains undefined reference to
                               `https://sevenbridges.commaxNumberOfParallelInstances`
INFO data_assembly_DNA_wf.cwl:90:3: Unknown hint https://sevenbridges.commaxNumberOfParallelInstances
WARNING Workflow checker warning:
data_assembly_DNA_wf.cwl:15:15: Source 'input_SV' of type ["null", "File"] may be incompatible
data_assembly_DNA_wf.cwl:55:7:    with sink 'input_SV' of type "File"
usage: data_assembly_DNA_wf.cwl [-h] --biospecimen_id_normal
                                BIOSPECIMEN_ID_NORMAL --biospecimen_id_tumor
                                BIOSPECIMEN_ID_TUMOR [--input_SV INPUT_SV]
                                --input_cnvkit_call_cns INPUT_CNVKIT_CALL_CNS
                                --input_cnvkit_call_seg INPUT_CNVKIT_CALL_SEG
                                --input_controlfreeC_info
                                INPUT_CONTROLFREEC_INFO
                                --input_controlfreeC_p_value
                                INPUT_CONTROLFREEC_P_VALUE --input_maf
                                INPUT_MAF --input_previous_merged_maf
                                INPUT_PREVIOUS_MERGED_MAF --participant_id
                                PARTICIPANT_ID --run_WGS
                                [job_order]
data_assembly_DNA_wf.cwl: error: the following arguments are required: --biospecimen_id_normal, --biospecimen_id_tumor, --input_cnvkit_call_cns, --input_cnvkit_call_seg, --input_controlfreeC_info, --input_controlfreeC_p_value, --input_maf, --input_previous_merged_maf, --participant_id, --run_WGS
```